### DISCUSSION TITLE
[bug] headershown: false still shows space

### DISCUSSION THREAD

**Discussion by ryanSoftwareEngineer** [react-navigation/react-navigation#11571] [General]
I have this screen that needs a drawer 

```
const SearchProfileDrawerNavigation = () => {
  return (
    <Drawer.Navigator
      screenOptions={{ drawerPosition: "right", swipeEdgeWidth: 0 }}
      drawerContent={(props) => <></>}
    >
      <Stack.Screen
        name={SEARCH_PROFILE_PAGE}
        key={"search" + SEARCH_PROFILE_PAGE}
        component={SearchProfileScreen}
        options={{ headerShown: true }}
        initialParams={{ isMain: false }}
      />
    </Drawer.Navigator>
  );
};

```

and outside there's a stack navigator 

```

const SearchNav = () => {
  console.log(" RENDERED search STACK NAVIGATOR ");
  return (
    <Stack.Navigator screenOptions={{ animation: "none", headerShown: false }}>
      <Stack.Screen
        name={SEARCH_PROFILE_WRAPPER}
        key={"search" + SEARCH_PROFILE_WRAPPER}
        component={SearchProfileDrawerNavigation}
        options={{
          headerShown: false,
          headerShadowVisible: false,

        }}
      />
    </Stack.Navigator>
  );
};

```

headerShown: false on the parent still shows a white blank space for the header

I ended up using the parent title and updating that but I don't like doing it that way because the component flow isn't the way my ocd wants it to be. 

Is there a more elegant way to work around this?

**Comment by AppLetico** [comment]
I am seeing the same issue, did you find a solution?

**Comment by YadBro** [comment]
+1 (2024)

**Comment by demacomsistemas** [comment]
Hi. I'm facing the same issue. Did you get a solution? (November 2024)

**Reply by YadBro** [reply]
Hi @demacomsistemas , I've found the solution (6.x.x version). The solution is you need to set **headerStatusBarHeight** to 0 in screenOptions props of Tab.Navigator: 
example:
<Tab.Navigator screenOptions={{headerStatusBarHeight: 0}} ></Tab.Navigator>

**Comment by demacomsistemas** [comment]
Hi @YadBro 
Excellent, glad to hear that.
You use Tab.Navigator... But I use a Stack Navigator and a Drawer Navigator inside it (or nested). I tried your solution, but can't make it work 


<Stack.Navigator screenOptions={{headerStatusBarHeight: 0, headerShown: false}}
            initialRouteName='Login'
        >
          <Stack.Screen name="Login" component={LoginScreen} options={{title: 'Login', fontFamily: 'CalibriRegular'}}/>
          <Stack.Screen name="EndpointConfig" component={EndpointConfigScreen} options={{title: 'Configuración', fontFamily: 'CalibriRegular'}} />
          <Stack.Screen name="Inicio" component={InicioScreen} options={{title: 'Principal', fontFamily: 'CalibriRegular'}} /> 
        </Stack.Navigator>

**Reply by YadBro** [reply]
What your react-navigation version?. may you can checkout this: https://snack.expo.dev/@yadiganteng32/drawer-navigator (this one using 7.0.0 version ~ new nov 2024)

**Comment by demacomsistemas** [comment]
These are my versions installed.

    "@react-navigation/drawer": "^7.0.1",
    "@react-navigation/native": "^7.0.0",
    "@react-navigation/stack": "^7.0.0",

**Reply by YadBro** [reply]
Hm..., try to open the issue again for up this issue and give them some to reproduce like using snack or whatever

**Comment by demacomsistemas** [comment]
@YadBro 
if I understand correctly, https://snack.expo.dev/@yadiganteng32/drawer-navigator is a drawer with a stack inside (the opposite of my case)... both with headerShown setted to false... I already tried to set headerShown to false in my both navigators, but no luck...

**Comment by demacomsistemas** [comment]
Well, I finally found my mistake. My home screen, which contains the drawer navigator, has a marginTop of 50. 
@YadBro thank you very much for your help

**Reply by YadBro** [reply]
Glad to hear 😊 🎉
